app/services/asset_manager/attachment_updater.rb

```diff
@@ -9,15 +9,25 @@ def self.call(
   )
     return if attachment_data.deleted?
 
-    updates = []
+    if attachment_data.assets.empty?
```

Comment:
Please add a comment to clarify that this is a feature flag and the code will eventually be removed!